examples/custom-server/server.js

```diff
@@ -0,0 +1,33 @@
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
+const accepts = require('accepts')
+
+const app = next('.', { dev: true })
+const handle = app.getRequestHandler()
+
+app.prepare()
+.then(() => {
+  createServer((req, res) => {
+    const accept = accepts(req)
+    const type = accept.type(['json', 'html'])
+    if (type === 'json') {
+      handle(req, res)
+      return
+    }
+
+    const { pathname, query } = parse(req.url, true)
+
+    if (pathname === '/a') {
+      app.render(req, res, '/b', query)
+    } else if (pathname === '/b') {
+      app.render(req, res, '/a', query)
+    } else {
```

Comment:
My temporary workaround is this: 

```javascript
let { pathname, query } = parse(req.url, true)

if (route('/blog/:id')(pathname).id) {
  query = query || {}
  query.params = route('/blog/:id')(pathname)
  app.render(req, res, '/blog__id', query)
} else {
  handle(req, res)
}
```
I feel like `app.render` should accept an optional `params` argument that gets passed into `getInitialProps(ctx)`?